### PR TITLE
SSP 4.99.162

### PR DESCRIPTION
**Pull request by cPanelChrisD**

### CODE REVIEW

File: ssp [hunk 1]
```diff
@@ -1378,6 +1385,16 @@ sub get_meminfo {
     return \%meminfo;
 }
 
+sub get_cpupdate_conf {
+    my %conf;
+    if ( open( my $conf_fh, '<', '/etc/cpupdate.conf' ) ) {
+        local $/ = undef;
+        %conf = map { ( split( /=/, $_, 2 ) )[ 0, 1 ] } split( /\n/, readline($conf_fh) );
```
Comment (davelcpanelnet-zz) [inline ssp:1392]: nice

File: ssp [hunk 2]
```diff
@@ -1755,20 +1762,7 @@ sub check_for_cpanel_update {
...
-            if (/\bcpanel=(.*)/i) {
-                $local_tier_name = $1;
-                last;
-            }
-        }
-        close $file_fh;
-    }
-
-    return if !$local_tier_name;
+    return unless defined $CPUPDATECONF->{CPANEL};
```
Comment (davelcpanelnet-zz) [inline ssp:1765]: Isn't $CPUPDATECONF->{CPANEL}; always going to be defined here? On line 1720 tht ternary is going to populate that scalar no matter what - just one way with the proper version and the other with a string representing an error state.  Would it be better to just populate the var on 1720 and move a conditional return here?

**Review by davelcpanelnet-zz — APPROVED** [review]
I don't see anything here worth redoing.  I did post some comments about line 1720 and how that feeds into line 1765 but that's really best practice bit - the data wanted comes through regardless and it's just opinion anyway!

Looks good!

File: ssp (hunk 2 above)
Comment (cPanelChrisD) [reply ssp:1765]: The ternary on 1720 won't define $CPUPDATECONF->{CPANEL}, it tests its defined-ness before using it, and does something else if it isn't.  1765 lies inside of a completely different function, so its defined-ness should be tested again.

**Pull request merged**